api/products_service.go

```diff
@@ -95,14 +96,13 @@ func (p ProductsService) Upload(input UploadProductInput) (UploadProductOutput,
 			case _ = <-respChan:
 				p.liveWriter.Stop()
 				return
+			case _ = <-progressDone:
+				p.progress.End()
+				p.liveWriter.Start()
+				liveLog = log.New(p.liveWriter, "", 0)
 			default:
```

Comment:
Does having this default case still result in a race condition? The tests end up breaking on the live-log updating because I request ends up in the `progressDone` state before ever kicking off `liveLog`. I would love for this not to be a default case and instead be checked explicitly.